java-extras/src/main/java/org/triplea/java/ColorUtils.java

```diff
@@ -0,0 +1,41 @@
+package org.triplea.java;
+
+import com.google.common.base.Preconditions;
+import java.awt.Color;
+import java.util.Random;
+import lombok.experimental.UtilityClass;
+
+@UtilityClass
+public class ColorUtils {
+  /**
+   * Returns a color parsed from the provided input hex string.
+   *
+   * @param colorString EG: 00FF00, FF00FF, 000000
+   */
+  public Color fromHexString(final String colorString) {
+    Preconditions.checkArgument(
+        colorString.length() == 6,
+        "Colors must be 6 digit hex numbers, eg FF0011, not: " + colorString);
+    try {
+      return new Color(Integer.decode("0x" + colorString));
+    } catch (final NumberFormatException nfe) {
+      throw new IllegalArgumentException(
+          "Colors must be 6 digit hex numbers, eg FF0011, not: "
+              + colorString
+              + ", "
+              + nfe.getMessage(),
+          nfe);
+    }
+  }
+
+  /** Returns a randomly generated color. */
+  public Color randomColor() {
```

Comment:
The name might be a bit misleading, it's more like an arbitrary number, also it doesn't look like this method is used at all.

Reply:
It was used, then I thought it made sense to keep the colors "sticky" so that games would be consistent. At a library/API level, I wouldn't want someone to use the `randomColor(long randomSeed)` API unless that is really what they wanted. I'm concerned someone would see that and either send a random seed because they had to or not use this API. My hope is that if you have a no-arg API, that would fit what you need. I don't think this is enough code to quite fit YAGNI given the library context, so I'm leaning we clean it up later if it's not used after a few months.

Reply:
In what sense is the name misleading, I'm not quite sure I see that. Could you help me understand that perspective?

Reply:
If I read "random" I expect a random result i.e. the result differs every time.
I think you can't tell that this method always returns the same arbitrary color from the name and javadoc.

Reply:
Ah, but be careful, the sequence of results will be the same if you restart the game, but that sequence will be random. If for example you request random colors in a different order for player names, the result would depend on the ordering of player names.

The other API is deterministic by player name because we use the player name hashcode for the random seed which will give us a fixed sequence for that player name (and we only grab the first element from that sequence).

Is it still your opinion that the method name should change or be clarified?

Reply:
I could be missing something, but right now this method has "123" hardcoded as the seed, so if it were ever to be called the color would always be the exact same

Reply:
Ah yeah, it would be exactly the same every time, that's a bug : )
I was thinking that the random generator would be new'ed up once rather than each time.

Reply:
Removed in: dc9d6b0d9